### PR TITLE
fix further panics in integer operations

### PR DESCRIPTION
**Pull request by davidhewitt**
Avoids more possible panics

---
## Summary by cubic
Prevents panics in integer-heavy paths (pow, range, bytes.hex, rsplit) and returns proper Python errors for edge cases. Behavior now matches CPython for modulus and argument parsing quirks.

- **Bug Fixes**
  - `pow(x, y, m)`: `m == 0` raises `ValueError`; `|m| == 1` returns `0`; negative exponent with a modulo raises `ValueError`; avoids `rem_euclid` overflow on `i64::MIN` with negative `m`.
  - `bytes.hex(sep, bytes_per_sep)`: parse `bytes_per_sep` as C `int`; out-of-range values raise `OverflowError: Python int too large to convert to C int`; prevent overflow when computing group size.
  - `range`: containment and slicing computed in `i128` to avoid overflow; slicing raises `OverflowError` when results can’t fit in `i64`; indexing guarded to return `OverflowError` instead of panicking.
  - `str.rsplit(maxsplit)`: correctly handles multi-byte whitespace and respects `maxsplit` without panics.
  - Added tests for these edge cases, including full `i64` spans and the intentional CPython divergence for overflowing `range` slices.

<sup>Written for commit 4c2cfa3d4a76b6e337c794e1d60bf8c8729e1509. Summary will update on new commits. <a href="https://cubic.dev/pr/pydantic/monty/pull/444?utm_source=github">Review in cubic</a></sup>

### CODE REVIEW

File: crates/monty/src/types/range.rs [hunk 1]
```diff
@@ -156,21 +156,26 @@ impl Range {
...
+        // The guarantee on `slice.indices` is that `stop` and `start` are at most
+        // `range_len` apart, so the subtraction won't overflow.
+        let num_elements = div_ceil(i128::from(stop) - i128::from(start), slice_step);
+        let new_stop_i128 = new_start_i128 + num_elements * new_step_i128;
 
-        // new_stop = new_start + num_elements * new_step
-        let new_stop = new_start.saturating_add(num_elements.saturating_mul(new_step));
+        let new_step = i64::try_from(new_step_i128).map_err(|_| ExcType::overflow_c_ssize_t())?;
+        let new_start = i64::try_from(new_start_i128).map_err(|_| ExcType::overflow_c_ssize_t())?;
+        let new_stop = i64::try_from(new_stop_i128).map_err(|_| ExcType::overflow_c_ssize_t())?;
```
[the remark this answers is not in the document]
Comment (davidhewitt) [reply crates/monty/src/types/range.rs:178]: The slice stores i64 internally so cannot repressent near-boundary slices. Maybe in future could store arbitrary-sized ints.
